File: src/main/java/nz/govt/linz/landonline/step/landonlite/controllers/TitlesController.java
package nz.govt.linz.landonline.step.landonlite.controllers;

import nz.govt.linz.landonline.step.landonlite.models.Title;
import nz.govt.linz.landonline.step.landonlite.repositories.TitleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Optional;

@RestController
@RequestMapping("/api/titles")
public class TitlesController {
    private static ResponseEntity NOT_FOUND = ResponseEntity.notFound().build();

    @Autowired
    private TitleRepository titleRepository;

    @GetMapping("/{id}")
    public ResponseEntity<Title> showTitle(@PathVariable long id) {
        Optional<Title> title = titleRepository.findById(id);
        return title.map(t -> ResponseEntity.ok().body(t)).orElse(NOT_FOUND);
    }

    @PostMapping("/{id}")
    public ResponseEntity<Title> updateTitle(@PathVariable long id, @RequestBody Title body) {
        Title result = titleRepository.findById(id).get();
        result.setOwnerName(body.getOwnerName());
        titleRepository.save(result);
        return ResponseEntity.ok().body(result);
    }
}

File: src/main/kotlin/nz/govt/linz/landonlite/controllers/TitlesController.kt
package nz.govt.linz.landonlite.controllers

import nz.govt.linz.landonlite.models.Title
import nz.govt.linz.landonlite.repositories.TitleRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/titles")
class TitlesController {
    @Autowired
    private lateinit var titleRepository: TitleRepository

    @GetMapping("/{id}")
    fun getTitle(@PathVariable id: Long): ResponseEntity<Title> {
        val title = titleRepository.findById(id)
        return title.map { t: Title -> ResponseEntity.ok().body(t) }
                .orElse(ResponseEntity.notFound().build())
    }

    @PostMapping("/{id}")
    fun updateTitle(@PathVariable id: Long, @RequestBody body: Title): ResponseEntity<Title> {
        val result = titleRepository.findById(id).get()
        result.ownerName = body.ownerName
        titleRepository.save(result)
        return ResponseEntity.ok().body(result)
    }
}